guides/hack/45-operators/05-pipe-operator.md

```diff
@@ -0,0 +1,62 @@
+The Pipe operator allows for a more concise, fluid syntax for chaining together expressions. It looks like this:
+
+```
+f() |> g($$) |> h(5, $$)
+```
+
+which conceptually means:
+
+```
+h(5, g(f()))
+```
+
+That is, evaluate the left hand side (LHS) of the pipe, then evaluate the RHS of the pipe with the result of the LHS bound to the placeholder `$$`.
+
+Runtime execution is explicitly left-to-right - a pipe's LHS is always executed before its RHS. More precisely, the LHS of a Pipe is evaluated and its result stored in a temporary, hidden variable. Then evaluate the RHS where `$$` refers to that hidden variable. This means you can use $$ more than once in the RHS but the LHS will still be evaluated only once. For example:
+
+```
+f() |> g1($$) + g2($$) |> h(5, $$)
+```
+
+roughly means:
+
+```
+h(g1(f()) + g2(f()))
+```
+
+except that `f()`:
+- is evaluated only once
+- is evaluated before either `g1` or `g2`.
+
+It may be easier to conceptualize this as a lambda rewrite:
```

Comment:
I think that's clearer. I don't think it's helping though :P 
This is more of the implementation the user needs to know. That lambdas don't have anything do to with the pipe operator makes this confusing to me.

All I needed to immediately understand how to use the operator and why it exists was the filter/map example from the proposal in https://github.com/facebook/hhvm/issues/6455
Especially the why one wants to use it is important for me. That's why I prefer examples showing real usage over function names with no meaning.

```
// map -> filter -> count
array(...)
  |> array_map($x ==> $x->getNumber(), $$)
  |> array_filter($$, $x ==> is_even($x))
  |> count($$)
```

The user doesn't really need to know about temporaries, scoping, and expression vs statement in my opinion. 

That's just my opinion though. I could be wrong ;-)